File: cpp/CF1023A.cpp
#include <bits/stdc++.h>
using namespace std;

int main()
{
	int n, m;
	string s, t;
	while (cin >> n >> m)
	{
		cin >> s >> t;
		int i = 0;
		for ( ; i < n; i++)
		{
			if (s[i] == '*') break;
		}
		if (i == n)
		{
			if (n == m && s == t) cout << "YES" << endl;
			else cout << "NO" << endl;
		}
		else
		{
			if (n > m + 1) { cout << "NO" << endl; continue; }
			string p = "", q = "";
			if (i != 0) p = s.substr(0, i);
			if (i != n - 1) q = s.substr(i + 1, n - i - 1);
			bool flg = true;
			if (p != "" && t.substr(0, i) != p) flg = false;
			if (q != "" && t.substr(m - (n - i - 1), n - i - 1) != q) flg = false;
			cout << (flg ? "YES" : "NO") << endl;
		}
	}
	return 0;
}

File: python3/CF1023A.py
n, m = list(map(int, input().split()))
s = input()
t = input()
if '*' not in s:
	ans = 'YES' if s == t else 'NO'
	print (ans)
elif n > m + 1:
	print ('NO')
else:
	p, q = s.split('*')
	flg = True
	if len(p) > 0 and t[:len(p)] != p: flg = False
	if len(q) > 0 and t[-len(q):] != q: flg = False
	ans = 'YES' if flg else 'NO'
	print (ans)